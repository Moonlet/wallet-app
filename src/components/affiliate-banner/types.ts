export const AffiliateBannerType = {
    LEDGER_NANO_X: {
        url: 'https://shop.ledger.com/pages/ledger-nano-x?r=69d534ac55d9',
        image: require('../../assets/images/png/affiliate-banners/ledger_nano_x.png')
    },
    UNSTOPPABLE_DOMAINS: {
        url: 'https://unstoppabledomains.com/r/moonlet',
        image: require('../../assets/images/png/affiliate-banners/unstoppabledomains.png')
    }
};

export interface IAffiliateBannerType {
    url: string;
    image: any;
}
