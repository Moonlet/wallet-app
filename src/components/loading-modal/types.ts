import { TransactionMessageType, TransactionMessageText } from '../../core/blockchain/types';

export interface ILoadingModalMessage {
    type: TransactionMessageType;
    text?: TransactionMessageText;
    component?: any; // React.ComponentType<any>;
}

export interface ILoadingModalState {
    isVisible: boolean;
    message: ILoadingModalMessage;
}
