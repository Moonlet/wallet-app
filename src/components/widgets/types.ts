import { Blockchain } from '../../core/blockchain/types';
import { handleCta } from '../../redux/ui/screens/data/handle-cta';
import { AccountType } from '../../redux/wallets/state';
import { IconValues } from '../icon/values';
import {
    clearScreenInputData,
    runScreenValidation,
    runScreenStateActions,
    setScreenInputData
} from '../../redux/ui/screens/input-data/actions';

export interface IScreenRequest {
    context: IScreenContext;
    user: IScreenUser;
}

export interface IScreenContext<ScreenParams = any> {
    screen: string;
    step?: string;
    tab?: string;
    flowId?: string;
    params?: ScreenParams;
    flowData?: any;
}

export interface IScreenCtaContextParams<F = any, S = any, E = any> {
    ctaId: string;
    flowInputData: F;
    screenInputData: S;
    extraParams?: E;
}

/** @deprecated use string instead */
export enum ContextScreen {
    DASHBOARD = 'dashboard',
    TOKEN = 'token',
    QUICK_STAKE_SELECT_VALIDATOR = 'quickStakeSelectValidator'
}

/** @deprecated use string instead */
export enum ContextTab {
    ACCOUNT = 'account'
}

export interface ISmartScreenActions {
    handleCta?: typeof handleCta;
    clearScreenInputData?: typeof clearScreenInputData;
    runScreenValidation?: typeof runScreenValidation;
    runScreenStateActions?: typeof runScreenStateActions;
    setScreenInputData?: typeof setScreenInputData;
}

export enum SmartScreenPubSubEvents {
    SCROLL_TO_END = 'scrollToEnd',
    COLLAPSE_ALL = 'collapseAll'
}

export interface IScreenUser {
    os: 'android' | 'ios' | 'web';
    deviceId: string;
    preferedCurrency: string;
    appVersion: string;
    theme: 'dark' | 'light';
    country?: string;
    lang: 'en';

    wallet: {
        pubKey: string;
        type: 'HD' | 'HW';
        hwOptions?: {
            vendor?: string;
            model?: string;
            connectionType?: string;
        };
    };

    blockchain: Blockchain;
    chainId: string;
    address: string;
    accountType: AccountType;
}

export interface IScreenResponse {
    widgets: IScreenWidget[];
    bottomFixedArea?: IScreenWidget;
    validation?: IScreenValidation;
    navigationOptions?: any;
}

export interface IScreenCtaResponse {
    cta: ICta;
}

export interface IScreenFieldValidation {
    fn: string;
    params?: any[];
    messages?: {
        [key: string]: {
            type: string;
            message: string;
        };
    };
}

export interface IScreenValidation {
    validators: {
        [field: string]: IScreenFieldValidation[];
    };
}

export interface ICtaAction<P = any> {
    type: 'callAction' | 'openUrl' | 'navigateTo' | 'onBack' | 'navigationOnPop';
    params: {
        action?: string;
        url?: string;
        count?: string;
        screen?: string;
        params?: P;
    };
}

export interface ICta {
    actions?: ICtaAction[];

    /** @deprecated use actions instead */
    type?: 'callAction' | 'openUrl' | 'navigateTo';

    /** @deprecated use actions instead */
    params?: {
        action?: string;
        url?: string;
        screen?: string;
        params?: any;
    };

    label?: string;

    buttonProps?: {
        primary?: boolean;
        secondary?: boolean;
        disabled?: boolean;
        disabledSecondary?: boolean;

        colors?: {
            /** @deprecated use textStyle instead */
            label?: string;

            bg?: string;
        };

        leftIcon?: IconValues;

        wrapperStyle?: any;
        buttonStyle?: any;
        textStyle?: any;
    };

    // TODO: check this
    screenDataValidation?: {
        context: IScreenContext; // not sure if needed
    };
}

export interface IScreenWidget {
    title?:
        | string
        | {
              value: string;
              containerStyle?: IDataStyle;
              textStyle?: IDataStyle;
              iconStyle?: IDataStyle;
          };
    expandable?: boolean;
    initialState?: 'collapsed' | 'expanded';
    style?: IDataStyle;
    modules: IScreenModule[];
    cta?: ICta;
    hideExpandableIcon?: boolean;
}

export interface IScreenModule {
    displayWhen?: 'collapsed' | 'expanded'; // if undefined, it will be displayed always
    hidden?: boolean;
    style?: IDataStyle;
    type:
        | ModuleTypes.ABSOLUTE_MODULES
        | ModuleTypes.AMOUNT_INPUT
        | ModuleTypes.AMOUNT_SELECTABLE_BOX
        | ModuleTypes.BALANCES_GRID_ICONS
        | ModuleTypes.CTA
        | ModuleTypes.GRADIENT_WRAPPER
        | ModuleTypes.ICON_ONE_LINE
        | ModuleTypes.ICON
        | ModuleTypes.ICON_TWO_LINES
        | ModuleTypes.IMAGE_BANNER
        | ModuleTypes.INPUT
        | ModuleTypes.TIMER_INTERVAL_PRICE_UPDATE
        | ModuleTypes.MD_TEXT
        | ModuleTypes.MODULE_COLUMNS_WRAPPER
        | ModuleTypes.MODULE_SELECTABLE_WRAPPER
        | ModuleTypes.MODULE_WRAPPER
        | ModuleTypes.ONE_LINE_TEXT_BANNER
        | ModuleTypes.PRICE_UPDATE
        | ModuleTypes.PROGRESS_BAR
        | ModuleTypes.SEPARATOR
        | ModuleTypes.SINGLE_BALANCE_ICON
        | ModuleTypes.STATIC_TEXT_COLUMNS_BOTTOM_HEADER
        | ModuleTypes.STATIC_TEXT_COLUMNS_TOP_HEADER
        | ModuleTypes.TEXT_LINE_ICON
        | ModuleTypes.THREE_LINES_CTA
        | ModuleTypes.THREE_LINES_ICON
        | ModuleTypes.TWO_LINES_TEXT_BANNER
        | ModuleTypes.VALIDATIONS;
    cta?: ICta;
    data:
        | I2LinesTextBannerData
        | I3LinesCtaData
        | IAmountInputData
        | IAmountSelectableBoxData
        | IBalanceGridData
        | IBalanceGridData[]
        | IGradientWrapperData
        | IIconData
        | IIconTwoLinesData
        | IIconOneLineData
        | IImageBannerData
        | IInputData
        | IMdTextData
        | IOneLineTextBannerData
        | IScreenModuleColumnsWrapperData
        | IScreenModuleSelectableWrapperData
        | IScreenModuleWrapperData
        | ISeparatorData
        | IStaticTextColumnData[]
        | IThreeLinesIconData;
    details?: any; // ex. amount, validatorId, ...
    info?: {
        position?: string; // 'top-right' is default
        style?: IDataStyle;
        data: IScreenModule;
    };

    state: {
        // uiStateSelector?: IStateSelector, // aka state modifier - not implemented now
        actions?: IStateSelector[];
        selectors: {
            [key: string]: IStateSelector;
        };
    };
}

export interface IStateSelector {
    fn: string;
    params?: any[];
}

export enum ModuleTypes {
    ABSOLUTE_MODULES = 'absolute-modules',
    AMOUNT_INPUT = 'amount-input',
    AMOUNT_SELECTABLE_BOX = 'amount-selectable-box',
    BALANCES_GRID_ICONS = 'balances-grid-icons',
    CTA = 'cta',
    GRADIENT_WRAPPER = 'gradient-wrapper',
    ICON_ONE_LINE = 'cta-one-line',
    ICON = 'icon',
    ICON_TWO_LINES = 'icon-two-lines',
    IMAGE_BANNER = 'image-banner',
    INPUT = 'input',
    TIMER_INTERVAL_PRICE_UPDATE = 'timer-interval-price-update',
    MD_TEXT = 'md-text',
    MODULE_COLUMNS_WRAPPER = 'module-columns-wrapper',
    MODULE_SELECTABLE_WRAPPER = 'module-selectable-wrapper',
    MODULE_WRAPPER = 'module-wrapper',
    ONE_LINE_TEXT_BANNER = 'one_line_text_banner',
    PRICE_UPDATE = 'price-update',
    PROGRESS_BAR = 'progress-bar',
    SEPARATOR = 'separator',
    SINGLE_BALANCE_ICON = 'single-balance-icon',
    STATIC_TEXT_COLUMNS_BOTTOM_HEADER = 'static-text-columns-bottom-header',
    STATIC_TEXT_COLUMNS_TOP_HEADER = 'static-text-columns-top-header',
    TEXT_LINE_ICON = 'text-line-icon',
    THREE_LINES_CTA = '3-lines-cta',
    THREE_LINES_ICON = 'three-lines-icon',
    TWO_LINES_TEXT_BANNER = '2-lines-text-banner',
    VALIDATIONS = 'validations'
}

/// Screen Module Data Types \\\

export interface IScreenModuleWrapperData {
    state: string; // DEFAULT | PENDING | HIDDEN
    stateModifierFn: string;
    data: {
        [state: string]: Partial<IScreenModule>;
    };
}

export interface IScreenModuleSelectableWrapperData {
    state: string; //
    stateModifierFn: string;
    style?: {
        [state: string]: IDataStyle;
    };
    submodules: IScreenModule[];
}

export interface IScreenModuleColumnsWrapperData {
    style?: IDataStyle;
    submodules: IScreenModule[];
}

export enum DataType {
    TEXT = 'TEXT',
    CURRENCY = 'CURRENCY'
}

export interface ITextData {
    value: string | number;
    params?: {
        [param: string]: IData;
    };
}

export interface ICurrencyData {
    value: string;
    symbol: string;
    decimals?: number;
    round?: {
        decimals: number;
        type?: 'UP' | 'DOWN';
    };
    beautify?: IBeautify;
}

export interface IBeautify {
    fromValue?: number;
    decimals?: number;
    symbol?: string;
    notation?: 'compact' | 'percent';
}

export interface IDataStyle {
    [key: string]:
        | string
        | number
        | {
              value: string | number;
              fn: 'normalize' | 'normalizeFontAndLineHeight' | 'gradient';
          };
}

export interface IData {
    type: DataType;
    style?: IDataStyle;
    data: ITextData | ICurrencyData;
}

export interface I3LinesCtaData {
    firstLine: IData[];
    secondLine: IData[];
    thirdLine: IData[];
}

// Used for `static-text-columns-top-header` and `static-text-columns-bottom-header`
export interface IStaticTextColumnData {
    header: IData[];
    body: IData[];
}

// Used for`balances-grid-icons` and `single-balance-icon`
export interface IBalanceGridData {
    balance: IData[];
    label?: IData[];
    icon: {
        value: IconValues | string;
        color: string;
    };
}

export interface IImageBannerData {
    image: IImageInfo;
}

export interface IImageInfo {
    url: string;
    width: number;
    height: number;
}

export interface IOneLineTextBannerData {
    line: IData[];
}

export interface I2LinesTextBannerData {
    firstLine: IData[];
    secondLine: IData[];
    icon?: {
        value: IconValues;
        color?: string;
    };
    backgroundColor?: string;
}

export interface ISeparatorData {
    color?: string;
}

export interface IIconTwoLinesData {
    firstLine: IData[];
    secondLine: IData[];
    icon?: {
        url: string;
        style?: IDataStyle;
    };
}

export interface IIconData {
    icon: IconValues;
}

export interface IPriceUpdateData {
    endpoint: {
        url: string;
        method: 'POST' | 'GET';
        data: any;
    };
    reduxKey: string;
    interval: number; // miliseconds
}

export interface ITimerUpdateData {
    numSeconds: number;
    reduxKey: string;
    endpoint: {
        url: string;
        method: 'POST' | 'GET';
        data: any;
    };
    cta: ICta;
    interval: number; // miliseconds
}

export interface IThreeLinesIconData {
    firstLine: IData[];
    secondLine: IData[];
    thirdLine: IData[];
    icon: {
        url: string;
        style?: IDataStyle;
    };
}

export interface IMdTextData {
    text: string;
    style?: any;
}

/**
 * value: +0.1 | half | all
 * percentage: 10%
 */
export interface IAmountInputAmountBox {
    type: 'value' | 'percentage';
    value: string | number;
    // label: string;
}

export interface IAmountInputData {
    input?: {
        style?: IDataStyle;
        textStyle?: IDataStyle;
    };
    labels?: IData[];
    amounts?: IAmountInputAmountBox[];
    placeholder?: {
        value: string;
        color?: string;
    };
    editable?: boolean;
    focus?: boolean;
    showValidations?: boolean;
    onChangeTextAction?: IStateSelector;
}

export interface IAmountSelectableBoxData {
    amounts: IAmountInputAmountBox[];
}

export interface IIconOneLineData {
    icon: {
        value: IconValues;
        style?: IDataStyle;
    };
    line: IData[];
}

export interface IInputData {
    style?: {
        input?: IDataStyle;
        inputContainer?: IDataStyle;
    };
    options?: {
        keyboardType?: string;
        placeholder?: {
            value: string;
            color?: string;
        };
        multiline?: boolean;
    };
}

export interface IGradientWrapperData {
    gradient: string[];
    submodules: IScreenModule[];
}

export interface IProgressBarData {
    percentage: number;
    backgroundStyle?: IDataStyle;
    barStyle?: IDataStyle;
}

export interface ITextLineIconData {
    icon: {
        value: IconValues;
        style?: IDataStyle;
    };
    line: IData[];
    lineStyle?: IDataStyle;
}

export interface IValidationData {
    fieldName?: string;
}

export interface IAbsoluteModulesData {
    module1: {
        module: IScreenModule;
        style?: IDataStyle;
    };
    module2: {
        module: IScreenModule;
        style?: IDataStyle;
    };
}
