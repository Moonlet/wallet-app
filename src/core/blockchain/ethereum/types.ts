export enum MethodSignature {
    INCREASE_ALLOWANCE = 'increaseAllowance(address,uint256)',
    DELEGATE = 'delegate(address,uint256)',
    UNDELEGATE = 'undelegate(address,uint256)',
    WITHDRAW_DELEGATED = 'withdrawDelegated(address,address)',
    TRANSFER = 'transfer(address,uint256)'
}
