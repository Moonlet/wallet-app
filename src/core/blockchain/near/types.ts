import BigNumber from 'bignumber.js';
import { PosBasicActionType } from '../types/token';

export enum NearTransactionActionType {
    TRANSFER = 'TRANSFER',
    FUNCTION_CALL = 'FUNCTION_CALL'
}

export enum NearFunctionCallMethods {
    DEPOSIT = 'deposit',
    STAKE = 'stake',
    UNSTAKE = 'unstake',
    WITHDRAW = 'withdraw'
}

export interface INearTransactionAction {
    type: NearTransactionActionType;
    params?: any[];
}

export interface INearTransactionAdditionalInfoType {
    currentBlockHash: string;
    actions: INearTransactionAction[];
    posAction?: PosBasicActionType;
    validatorName?: string;
}

export enum NearAccountType {
    DEFAULT = 'DEFAULT',
    CONTRACT = 'CONTRACT'
}

export interface INearAccount {
    address: string;
    name: string;
    amount: BigNumber;
    exists: boolean;
    valid: boolean;
    type?: NearAccountType;
}
