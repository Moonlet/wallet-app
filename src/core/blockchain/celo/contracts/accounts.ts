import { Client } from '../client';
import abi from 'ethereumjs-abi';
import { getContract, buildBaseTransaction } from './base-contract';
import { Contracts } from '../config';
import { IPosTransaction, IBlockchainTransaction, TransactionType } from '../../types';
import { PosBasicActionType, TokenType } from '../../types/token';

export class Accounts {
    constructor(private client: Client) {}

    public async isRegisteredAccount(accountAddress: string): Promise<boolean> {
        const contractAddress = await getContract(this.client.chainId, Contracts.ACCOUNTS);

        return this.client
            .callContract(contractAddress, 'isAccount(address):(bool)', [accountAddress])
            .then(res => {
                if (res === 'false') return false;
                return true;
            })
            .catch(() => {
                return false;
            });
    }

    public async createAccount(tx: IPosTransaction): Promise<IBlockchainTransaction> {
        const transaction = await buildBaseTransaction(tx);
        const contractAddress = await getContract(this.client.chainId, Contracts.ACCOUNTS);

        const raw = '0x' + abi.simpleEncode('createAccount()').toString('hex');

        const fees = await this.client.getFees(
            TransactionType.CONTRACT_CALL,
            {
                from: tx.account.address,
                to: tx.account.address,
                amount: tx.amount,
                contractAddress,
                raw
            },
            TokenType.ERC20
        );
        transaction.feeOptions = fees;

        transaction.toAddress = contractAddress;
        transaction.amount = '0';
        transaction.data = {
            method: 'register',
            params: [tx.account.address],
            raw
        };

        transaction.additionalInfo.posAction = PosBasicActionType.CREATE_ACCOUNT;

        return transaction;
    }
}
