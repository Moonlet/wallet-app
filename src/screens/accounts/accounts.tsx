import React, { useState, useEffect } from 'react';
import { View } from 'react-native';
import { IReduxState } from '../../redux/state';
import { withTheme, IThemeProps } from '../../core/theme/with-theme';
import { smartConnect } from '../../core/utils/smart-connect';
import { connect } from 'react-redux';
import { IWalletState, IAccountState, TokenType } from '../../redux/wallets/state';
import { switchSelectedAccount, getBalance } from '../../redux/wallets/actions';
import stylesProvider from './styles';
import { Blockchain } from '../../core/blockchain/types';
import { selectCurrentWallet, getCurrentAccount } from '../../redux/wallets/selectors';
import { formatAddress } from '../../core/utils/format-address';
import { Text } from '../../library';
import { Amount } from '../../components/amount/amount';
import { getBlockchain } from '../../core/blockchain/blockchain-factory';
import { ListAccount } from '../token/components/list-account/list-account';
import BigNumber from 'bignumber.js';

export interface IReduxProps {
    wallet: IWalletState;
    switchSelectedAccount: typeof switchSelectedAccount;
    selectedAccount: IAccountState;
    getBalance: typeof getBalance;
    exchangeRates: any;
}

const mapStateToProps = (state: IReduxState) => {
    return {
        wallet: selectCurrentWallet(state),
        selectedAccount: getCurrentAccount(state),
        exchangeRates: (state as any).market.exchangeRates
    };
};

const mapDispatchToProps = {
    switchSelectedAccount,
    getBalance
};

// TODO: this is a component, not a screen any more => move this

export const AccountsScreenComponent = (
    props: IReduxProps & IThemeProps<ReturnType<typeof stylesProvider>>
) => {
    const [accounts, setAccounts] = useState([]);

    useEffect(() => {
        (async function getAccounts() {
            try {
                const hdAccounts = props.wallet.accounts.filter(
                    (account: IAccountState) =>
                        account.blockchain === props.selectedAccount.blockchain
                );
                setAccounts(hdAccounts);

                hdAccounts.map(account =>
                    this.props.getBalance(account.blockchain, account.address, undefined, true)
                );
            } catch (e) {
                // console.log(e);
            }
        })();
    }, [props.selectedAccount.blockchain]);

    const calculateBalance = (account: IAccountState) => {
        const tokenKeys = Object.keys(account.tokens);
        let balance = new BigNumber(0);
        tokenKeys.map(key => {
            const token = account.tokens[key];
            if (token.active) {
                if (token.type === TokenType.NATIVE) {
                    balance = balance.plus(token.balance?.value);
                } else {
                    const exchange =
                        props.exchangeRates[key][getBlockchain(account.blockchain).config.coin];
                    balance = balance.plus(token.balance?.value.multipliedBy(exchange));
                }
            }
        });
        return balance;
    };

    return (
        <View style={props.styles.container}>
            {accounts.map((account: IAccountState, index: number) => {
                const selected = props.selectedAccount.address === account.address;
                const blockchain = account.blockchain;

                const label = (
                    <View>
                        <View style={props.styles.firstRow}>
                            <Text style={props.styles.accountName}>
                                {`Account ${account.index + 1}`}
                            </Text>
                            <Text style={props.styles.accountAddress}>
                                {formatAddress(account.address)}
                            </Text>
                        </View>
                        <View style={{ flexDirection: 'row', alignItems: 'baseline' }}>
                            <Amount
                                style={props.styles.fistAmountText}
                                amount={calculateBalance(account)}
                                blockchain={blockchain}
                                token={getBlockchain(blockchain).config.coin}
                                tokenDecimals={
                                    getBlockchain(blockchain).config.tokens[
                                        getBlockchain(blockchain).config.coin
                                    ].decimals
                                }
                            />
                            <Amount
                                style={props.styles.secondAmountText}
                                amount={calculateBalance(account)}
                                blockchain={blockchain}
                                token={getBlockchain(blockchain).config.coin}
                                tokenDecimals={
                                    getBlockchain(blockchain).config.tokens[
                                        getBlockchain(blockchain).config.coin
                                    ].decimals
                                }
                                convert
                            />
                        </View>
                    </View>
                );
                return (
                    <ListAccount
                        key={index}
                        leftIcon={
                            blockchain === Blockchain.ETHEREUM
                                ? require('../../assets/images/png/eth.png')
                                : blockchain === Blockchain.ZILLIQA
                                ? require('../../assets/images/png/zil.png')
                                : undefined
                        }
                        rightIcon={selected ? 'check-1' : undefined}
                        label={label}
                        selected={selected}
                        onPress={() =>
                            props.switchSelectedAccount({
                                index: account.index,
                                blockchain
                            })
                        }
                    />
                );
            })}
        </View>
    );
};

export const AccountsScreen = smartConnect(AccountsScreenComponent, [
    connect(mapStateToProps, mapDispatchToProps),
    withTheme(stylesProvider)
]);
