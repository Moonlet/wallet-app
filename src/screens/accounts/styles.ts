import { StyleSheet } from 'react-native';
import { ITheme } from '../../core/theme/itheme';
import { BASE_DIMENSION } from '../../styles/dimensions';

export default (theme: ITheme) =>
    StyleSheet.create({
        container: {
            padding: BASE_DIMENSION * 2,
            ...StyleSheet.absoluteFillObject
        }
    });
