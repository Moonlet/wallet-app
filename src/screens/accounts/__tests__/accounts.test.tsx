import React from 'react';
import { shallow } from 'enzyme';
import { AccountsScreenComponent, IProps, IReduxProps, INavigationParams } from '../accounts';
import stylesProvider from '../styles';
import { darkTheme } from '../../../styles/themes/dark-theme';
import { loadTranslations } from '../../../core/i18n';
import { WalletType } from '../../../core/wallet/types';
import { IThemeProps } from '../../../core/theme/with-theme';
import { Blockchain } from '../../../core/blockchain/types';
import { INavigationProps } from '../../../navigation/with-navigation-params';

const props: IReduxProps &
    INavigationProps<INavigationParams> &
    IThemeProps<ReturnType<typeof stylesProvider>> = {
    // @ts-ignore
    navigation: {
        navigate: jest.fn()
    },
    blockchain: Blockchain.ETHEREUM,
    styles: stylesProvider(darkTheme),
    wallet: {
        id: 'walletId',
        type: WalletType.HD,
        name: 'some',
        accounts: []
    },
    addAccount: jest.fn(),
    removeAccount: jest.fn()
};

export default describe('Accounts screen', () => {
    beforeAll(async () => {
        await loadTranslations('en');
    });

    test('renders correctly', () => {
        const wrapper = shallow(<AccountsScreenComponent {...props} />);
        expect(wrapper.debug()).toMatchSnapshot();
    });
});
