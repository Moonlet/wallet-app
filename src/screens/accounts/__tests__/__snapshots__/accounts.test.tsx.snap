// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accounts screen renders correctly 1`] = `"<View style={{...}} />"`;
